### PR TITLE
Remove containerd deprecated registry options

### PR DESCRIPTION
**Pull request by adisky**
remove containerd deprecated registry configs

update plugins."io.containerd.grpc.v1.cri".registry.mirrors-->plugins.'"io.containerd.grpc.v1.cri'".registry config_path

ref: https://github.com/containerd/containerd/pull/9766

### CODE REVIEW

**Comment by adisky** [comment]
/hold
will create a new containerd 2.0 config and test it for one job , then update the currently used config which is used by many jobs to have minimal impact in case something fails

**Comment by BenTheElder** [comment]
https://github.com/kubernetes/test-infra/issues/31851

File: jobs/e2e_node/containerd/hosts-k8s.toml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+server = "https://k8s.gcr.io"
```
Comment (BenTheElder) [inline jobs/e2e_node/containerd/hosts-k8s.toml:1]: we really shouldn't need this anymore, jobs should not be pulling from k8s.gcr.io, if they still are then that's a bug.

Comment (adisky) [reply jobs/e2e_node/containerd/hosts-k8s.toml:1]: will remove this

**Comment by adisky** [comment]
Migrating single job to use containerd config_path

**Comment by AkihiroSuda** [comment]
What's the current status?

**Comment by adisky** [comment]
we have migrated one job and it is running fine
https://github.com/kubernetes/test-infra/pull/31954
https://github.com/kubernetes/test-infra/pull/31970

**Comment by adisky** [comment]
cc @SergeyKanzhelev

**Review by samuelkarp — APPROVED** [review]

**Review by ruiwen-zhao — APPROVED** [review]

File: jobs/e2e_node/containerd/image-config-serial-eviction.yaml [hunk 2]
```diff
@@ -3,4 +3,7 @@ images:
     image_family: cos-beta
     project: cos-cloud
     machine: n1-standard-2 # These tests need a lot of memory
-    metadata: "user-data</home/prow/go/src/k8s.io/test-infra/jobs/e2e_node/containerd/init-eviction.yaml,cni-template</home/prow/go/src/k8s.io/test-infra/jobs/e2e_node/containerd/cni.template,containerd-config</home/prow/go/src/k8s.io/test-infra/jobs/e2e_node/containerd/config-systemd.toml"
+    metadata: "user-data</home/prow/go/src/k8s.io/test-infra/jobs/e2e_node/containerd/init-eviction.yaml,cni-template</home/prow/go/src/k8s.io/test-infra/jobs/e2e_node/containerd/cni.template,containerd-config</home/prow/go/src/k8s.io/test-infra/jobs/e2e_node/containerd/config-systemd.toml,registry-config-docker</home/prow/go/src/k8s.io/test-infra/jobs/e2e_node/containerd/hosts-docker.toml"
+  cos-stable:
+    image_family: cos-stable
+    project: cos-cloud"
```
Comment (SergeyKanzhelev) [inline jobs/e2e_node/containerd/image-config-serial-eviction.yaml:9]: is end quote here needed?

File: jobs/e2e_node/containerd/image-cadvisor.yaml [hunk 3]
```diff
@@ -2,8 +2,13 @@ images:
   ubuntu:
     image_family: pipeline-1-29
     project: ubuntu-os-gke-cloud
-    metadata: "user-data</home/prow/go/src/k8s.io/test-infra/jobs/e2e_node/containerd/init.yaml,cni-template</home/prow/go/src/k8s.io/test-infra/jobs/e2e_node/containerd/cni.template,containerd-config</home/prow/go/src/k8s.io/test-infra/jobs/e2e_node/containerd/config.toml"
+    metadata: "user-data</home/prow/go/src/k8s.io/test-infra/jobs/e2e_node/containerd/init.yaml,cni-template</home/prow/go/src/k8s.io/test-infra/jobs/e2e_node/containerd/cni.template,containerd-config</home/prow/go/src/k8s.io/test-infra/jobs/e2e_node/containerd/config.toml,registry-config-docker</home/prow/go/src/k8s.io/test-infra/jobs/e2e_node/containerd/hosts-docker.toml"
+    image_family: cos-stable
+    project: cos-cloud"
```
Comment (ruiwen-zhao) [inline jobs/e2e_node/containerd/image-cadvisor.yaml:7]: extra quote at the end

File: jobs/e2e_node/containerd/image-cadvisor.yaml [hunk 4]
```diff
@@ -2,8 +2,13 @@ images:
...
   cos-stable:
     image_family: cos-beta
     project: cos-cloud
-    metadata: "user-data</home/prow/go/src/k8s.io/test-infra/jobs/e2e_node/containerd/init.yaml,cni-template</home/prow/go/src/k8s.io/test-infra/jobs/e2e_node/containerd/cni.template,containerd-config</home/prow/go/src/k8s.io/test-infra/jobs/e2e_node/containerd/config.toml"
+    metadata: "user-data</home/prow/go/src/k8s.io/test-infra/jobs/e2e_node/containerd/init.yaml,cni-template</home/prow/go/src/k8s.io/test-infra/jobs/e2e_node/containerd/cni.template,containerd-config</home/prow/go/src/k8s.io/test-infra/jobs/e2e_node/containerd/config.toml,registry-config-docker</home/prow/go/src/k8s.io/test-infra/jobs/e2e_node/containerd/hosts-docker.toml"
+  cos-stable:
+    image_family: cos-stable
+    project: cos-cloud"
```
Comment (ruiwen-zhao) [inline jobs/e2e_node/containerd/image-cadvisor.yaml:14]: extra quote at the end

**Comment by adisky** [comment]
@ruiwen-zhao @SergeyKanzhelev Updated

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve

**Pull request merged**